File: backend/app/api/v1/routes/market_context.py
from typing import Annotated

from fastapi import APIRouter, Depends

from app.modules.market_context.dependencies import get_market_context_service
from app.modules.market_context.schemas import MarketContextResponse
from app.modules.market_context.service import MarketContextService

router = APIRouter(prefix="/assets", tags=["market-context"])
type Service = Annotated[MarketContextService, Depends(get_market_context_service)]


@router.get("/{symbol}/market-context", response_model=MarketContextResponse)
async def market_context(symbol: str, service: Service) -> MarketContextResponse:
    return await service.get_context(symbol)

File: frontend/src/lib/api/market-context.ts
import { apiGet } from "./client";
import type { MarketContext } from "./market-context-types";

export const marketContextApi = {
  get: (symbol: string, signal?: AbortSignal) =>
    apiGet<MarketContext>(
      `/assets/${encodeURIComponent(symbol)}/market-context`,
      signal,
    ),
};
